From 213b6cd9fb24381563d6d44c7ddc4700abfae676 Mon Sep 17 00:00:00 2001
From: Vladimir L Ivanov <wawa@yandex-team.ru>
Date: Thu, 21 Oct 2010 14:59:54 +0400
Subject: [PATCH] bgpd: fix printed value of last-update timestamp

* bgp_route.c: (route_vty_out_detail) calculate time value
    in a way, which works regardless of monotonic clock
    being used or not
---
 bgpd/bgp_route.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

--- a/bgpd/bgp_route.c
+++ b/bgpd/bgp_route.c
@@ -6145,7 +6145,8 @@ route_vty_out_detail (struct vty *vty, s
 	bgp_damp_info_vty (vty, binfo);
 
       /* Line 7 display Uptime */
-      vty_out (vty, "      Last update: %s", ctime (&binfo->uptime));
+      time_t tbuf = time(NULL) - (bgp_clock() - binfo->uptime);
+      vty_out (vty, "      Last update: %s", ctime(&tbuf));
     }
   vty_out (vty, "%s", VTY_NEWLINE);
 }  
